java-checks/src/main/java/org/sonar/java/checks/LeastSpecificTypeCheck.java

```diff
@@ -204,12 +201,30 @@ private Type leastSpecificType() {
       return longestChain.get(0);
     }
 
-    private static boolean isOverriding(Symbol s1, Symbol s2, ClassJavaType superClass) {
-      return s1.isMethodSymbol() && s2.isMethodSymbol() && isOverridingMethod(((JavaSymbol.MethodJavaSymbol) s1), ((JavaSymbol.MethodJavaSymbol) s2), superClass);
+    private static boolean isOverriding(Symbol s1, Symbol s2) {
+      return s1.isMethodSymbol()
+        && s2.isMethodSymbol()
+        && s1.name().equals(s2.name())
+        && isPotentialOverride((Symbol.MethodSymbol) s1, (Symbol.MethodSymbol) s2);
     }
 
-    private static boolean isOverridingMethod(JavaSymbol.MethodJavaSymbol s1, JavaSymbol.MethodJavaSymbol s2, ClassJavaType superClass) {
-      return s1.name().equals(s2.name()) && BooleanUtils.isTrue(s1.checkOverridingParameters(s2, superClass));
+    private static boolean isPotentialOverride(Symbol.MethodSymbol method, Symbol.MethodSymbol overrideeCandidate) {
```

Comment:
This is a pure duplicate of method in another class. If we don't want to  add method to some helper class, then let's at least reuse the same method, since it is static.